### PR TITLE
getting started 2.x

### PR DESCRIPTION
**Pull request by jamiltz**
Copied from https://developer.couchbase.com/documentation/server/5.0/connectors/elasticsearch-2.2/install-and-config.html so it's probably a straightforward merge.

### CODE REVIEW

File: docs/getting-started.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+4. Download and install the plug-in package.
+
+	```bash
+	$ bin/plugin install https://github.com/couchbaselabs/elasticsearch-transport-couchbase/releases/download/3.0.0-cypress/elasticsearch-transport-couchbase-3.0.0-alder-es2.4.6.zip
+	```
+
+When the installation is successful, the message "Installed transport-couchbase into /usr/share/elasticsearch/plugins/transport/couchbase" is logged.
+
+5. Open the Elasticsearch configuration file (**/etc/elasticsearch/elasticsearch.yml**) and add the following to the end of the file. The **username** and **password** should be the one used to connect to Couchbase Server.
```
Comment (dnault) [inline docs/getting-started.md]: The text about username and password is misleading. What's actually happening is that here in the Elasticseach config the user is *defining* the credentials that must be used when configuring XDCR. When Couchbase makes an XDCR request to the ES plugin, the plugin will check the credentials against the ones defined here in the ES config, and will reject the XDCR request if they don't match.

That said, the next major version of the ES plugin will likely use a different replication scheme, and the username and password *will* need to match the credentials of a Couchbase user with read access to the bucket being replicated.

File: docs/getting-started.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+7. Still in the **/usr/share/elasticsearch** directory, configure the Elasticsearch plugin with the following curl command (Couchbase Server must be running).
+
+	```bash
+	$ curl -X PUT http://localhost:9200/_template/couchbase -d @plugins/transport-couchbase/couchbase_template.json
+	```
+
+	When successful, the configuration-routine provides the following response: `{"acknowledged":true}`.
+
+8. Instantiate an Elasticsearch index to be applied by Elasticsearch to the data in a given Couchbase bucket (in this case, the **travel-sample** bucket).
```
Comment (dnault) [inline docs/getting-started.md]: I'd rephrase this along the lines of: "Create an Elasticsearch index to receive the data from the Couchbase bucket. Later, when configuring XDCR, the "remote bucket name" must match the Elasticsearch index name."

File: docs/getting-started.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+	```bash
+	$ curl -X PUT http://localhost:9200/travel-sample
+	
+	{"acknowledged":true}
+	```
+
+You have now completed all basic installation and configuration requirements for Elasticsearch, the Elasticsearch Plug-in, and Couchbase Server.
```
Comment (dnault) [inline docs/getting-started.md]: This guide doesn't really cover installing Couchbase Server... and it probably doesn't need to.

It should probably talk about setting up XDCR though, emphasizing that:

1) When configuring the "Remote Cluster" (actually ES), the username and password must match the credentials defined in the ES config. The IP/Hostname must specify the port the ES plugin is listening on (default: 9091)
2)  When creating the replication, you must select "Show Advanced Settings" and change the XDCR Protocol to "Version 1", otherwise you'll get an unhelpful error messaged from ES.
3) The "remote bucket" in the XDCR config must be the ES index name (which may be different from the couchbase bucket name).
4) For best performance, the optimistic replication threshold in the XDCR config should be set to the maximum value of 20971520.

File: docs/getting-started.md [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+This section provides a step-by-step procedure, whereby the three components can be installed, configured, and run.
+
+## Pre-requisites
+
+- Ubuntu 14.04 or 16.04
+- Java
+
+## Installation
+
+1. Download the Elasticsearch 2.4.0 package for Ubuntu.
```
Comment (dnault) [inline docs/getting-started.md]: Anyone who's using ES 2.x already has it installed. For new users, we should recommend the latest version of 5.x.

Perhaps we should also mention that ES 6.0 is not supported yet.

**Comment by jamiltz** [comment]
@dnault how can i tell that the plugin is indeed running? should it be with `curl localhost:8091`?

Here's what I get with `curl -X PUT http://localhost:9200/travel-sample`

```json
{"error":{"root_cause":[{"type":"index_already_exists_exception","reason":"index [travel-sample/dyn_qksNSkGtiA7lMgWMyA] already exists","index_uuid":"dyn_qksNSkGtiA7lMgWMyA","index":"travel-sample"}],"type":"index_already_exists_exception","reason":"index [travel-sample/dyn_qksNSkGtiA7lMgWMyA] already exists","index_uuid":"dyn_qksNSkGtiA7lMgWMyA","index":"travel-sample"},"status":400}
```

**Comment by dnault** [comment]
Just putting this here for posterity. You can verify the plugin is running with

    curl <username>:<password>@localhost:9091

It should return: `{"welcome":"elasticsearch-transport-couchbase"}`

File: docs/getting-started.md [hunk 5]
```diff
@@ -3,73 +3,97 @@
...
+couchbase.maxConcurrentRequests: 1024
+```
 
-	```bash
-	$ bin/plugin install https://github.com/couchbaselabs/elasticsearch-transport-couchbase/releases/download/3.0.0-cypress/elasticsearch-transport-couchbase-3.0.0-alder-es2.4.6.zip
-	```
+The **username** and **password** credentials will be used again later when configuring Couchbase Server.
 
-When the installation is successful, the message "Installed transport-couchbase into /usr/share/elasticsearch/plugins/transport/couchbase" is logged.
+Still in the **/usr/share/elasticsearch** directory, configure the Elasticsearch plugin with the following curl command (Couchbase Server must be running).
```
Comment (dnault) [inline docs/getting-started.md]: I don't think Couchbase needs to be up for this step. Do you mean, "Elasticsearch server must be running"?

File: docs/getting-started.md [hunk 6]
```diff
@@ -3,73 +3,97 @@
...
-	```bash
-	$ curl -X PUT http://localhost:9200/_template/couchbase -d @plugins/transport-couchbase/couchbase_template.json
-	```
+Now to configure Couchbase Server, open the Couchbase Web Console and select **XDCR > Add Remote Cluster**.
 
-	When successful, the configuration-routine provides the following response: `{"acknowledged":true}`.
+![](img/xdcrInitial.png)
 
-8. Instantiate an Elasticsearch index to be applied by Elasticsearch to the data in a given Couchbase bucket (in this case, the **travel-sample** bucket).
+In the dialog, enter the **Cluster Name** of your choice, the **IP/hostname** should correspond to the Elasticsearch cluster and **Username**/**Password** to the credentials previously stored in **elasticsearch.yml**.
```
Comment (dnault) [inline docs/getting-started.md]: Might be good to emphasize the port number 9091 needs to be specified, since by default the ES plugin does not listen on the standard XDCR port.

**Comment by jamiltz** [comment]
Changes per feedback:

- [x] unhyphenated "Couchbase Server cluster" over "Server-cluster".
- [x] remove ubuntu as a pre-requisite and direct users to ES docs.
- [x] mention the plugin doesn't support ES6.
- [x] recommend to install 5.6.4 of the plugin.
- [x] clarify the role of elastic search credentials.
- [x] point out port 9091 is the default for the plugin.

File: docs/getting-started.md [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+
+In the dialog, enter the **Cluster Name** of your choice, the **IP/hostname** and port number where the Elasticsearch cluster is running (the Elasticsearch plugin listens on port 9091 by default) and the **Username**/**Password** previously stored in **elasticsearch.yml**.
+
+![](assets/remotecluster.png)
+
+Next, select the **Add Replication** option.
+
+![](assets/addreplication.png)
+
+In the pop-up window, enter the origin bucket, the remote cluster and remote bucket (in this case it's the index created earlier). Also make sure that the **XDCR Protocol** version is set to 1 and click **Save**.
```
Comment (dnault) [inline docs/getting-started.md]: NIT: Oxford comma between "remote cluster" and "remote bucket"?

Comment (dnault) [inline docs/getting-started.md]: Setting the "XDCR Optimistic Replication Threshold" to the max value of 20971520 can have such a great impact on performance that it's probably worth including it in the basic setup instructions. Don't need to make an additional image (unless you want to); a simple mention would probably be enough.

**Comment by jamiltz** [comment]
Updated per review.

**Comment by dnault** [comment]
Awesome, thanks!
